targets/TARGET_TOSHIBA/TARGET_TMPM066/analogin_api.c

```diff
@@ -0,0 +1,89 @@
+/* mbed Microcontroller Library
+ * (C)Copyright TOSHIBA ELECTRONIC DEVICES & STORAGE CORPORATION 2017 All rights reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "analogin_api.h"
+#include "PeripheralNames.h"
+#include "pinmap.h"
+#include "mbed_wait_api.h"
+
+#define ADC_10BIT_RANGE        0x3FF
+
+static const PinMap PinMap_ADC[] = {
+    {PA0, ADC_A0, PIN_DATA(0, 0)},
+    {PA1, ADC_A1, PIN_DATA(0, 0)},
+    {PA2, ADC_A2, PIN_DATA(0, 0)},
+    {PA3, ADC_A3, PIN_DATA(0, 0)},
+    {PA4, ADC_A4, PIN_DATA(0, 0)},
+    {PA5, ADC_A5, PIN_DATA(0, 0)},
+    {PA6, ADC_A6, PIN_DATA(0, 0)},
+    {PA7, ADC_A7, PIN_DATA(0, 0)},
+    {NC,  NC,     0}
+};
+
+void analogin_init(analogin_t *obj, PinName pin)
+{
+    // Check that pin belong to ADC module
+    obj->adc = (ADCName)pinmap_peripheral(pin, PinMap_ADC);
+    MBED_ASSERT(obj->adc != (ADCName)NC);
+
+    // enable clock supply to ADC
+    CG_SetFcPeriphA(CG_FC_PERIPH_ADC, ENABLE);
+    CG_SetADCClkSupply(ENABLE);
+    // Set pin function as ADC
+    pinmap_pinout(pin, PinMap_ADC);
+    // Software reset ADC
+    ADC_SWReset();
```

Comment:
A software reset will initializes all the registers again (even pre-configured ADC settings), except clock register. This is fixed to avoid conflict with some already configured ADC pins and committed the changes.